cmd/experimental/kjobctl/pkg/builder/builder.go

```diff
@@ -236,18 +247,36 @@ func (b *Builder) buildPodSpec(templateSpec corev1.PodSpec) corev1.PodSpec {
 
 		container.VolumeMounts = append(container.VolumeMounts, bundle.Spec.ContainerVolumeMounts...)
 		container.Env = append(container.Env, bundle.Spec.EnvVars...)
+		container.Env = append(container.Env, b.additionalEnvironmentVariables()...)
 	}
 
 	for i := range templateSpec.InitContainers {
 		initContainer := &templateSpec.InitContainers[i]
 
 		initContainer.VolumeMounts = append(initContainer.VolumeMounts, bundle.Spec.ContainerVolumeMounts...)
 		initContainer.Env = append(initContainer.Env, bundle.Spec.EnvVars...)
+		initContainer.Env = append(initContainer.Env, b.additionalEnvironmentVariables()...)
 	}
 
 	return templateSpec
 }
 
+func (b *Builder) additionalEnvironmentVariables() []corev1.EnvVar {
+	userID := os.Getenv(constants.SystemEnvVarNameUser)
+	timestamp := b.buildTime.Format(time.RFC3339)
+	taskName := fmt.Sprintf("%s_%s", b.namespace, b.profileName)
+
+	envVars := []corev1.EnvVar{
+		{Name: constants.EnvVarNameUserID, Value: userID},
+		{Name: constants.EnvVarTaskName, Value: taskName},
+		{Name: constants.EnvVarTaskID, Value: fmt.Sprintf("%s_%s_%s", userID, timestamp, taskName)},
+		{Name: constants.EnvVarNameProfile, Value: fmt.Sprintf("%s_%s", b.namespace, b.profileName)},
```

Comment:
@mwielgus @alculquicondor Also I have a question about namespace prefix. In documentation we have `TASK_NAME - namespace +”_”+ name.
`. Maybe it's better to create another variable like `NAMESPACE`. In this case user can combine it if it's need. WDYT?

Reply:
Namespace, if needed is available via K8S downward api.